docs/e2e.md

```diff
@@ -0,0 +1,234 @@
+    accessmodes: ReadWriteOnce
+```
+
+### Explanation of Config Fields
+
+- `channelname`:
+    Name of the channel for the channel CR. The channel CR
+    is later used by subscription CR.
+
+- `channelnamespace`:
+    Namespace in which the channel CR is created.
+
+- `giturl`:
+    Git URL pointing to the repository with application manifests.
+
+    Deployment uses below fixed values, and git url defined above,
+    pointing to an application in the git url to be deployed on the cluster:
+
+    ```
+    GITPATH     = "workloads/deployment/base"
+    GITREVISION = "main"
+    APPNAME     = "busybox"
+    ```
+
+- `pvcspecs`:
+    are used to kustomize the workload to use different pvc
+    parameters like `storageclassname` and `accessmodes`.
+
+    Each pvc specification in the list of **pvcspecs** will be applied
+    for each workload. By default, it will creates two busybox deployments
+    with different pvc parameters, one with CephFS and RWX and the other
+    with RBD and RWO.
+
+## Prepare `config.yaml` File
+
+- Create a config.yaml file by copying `config.yaml.sample` template:
+
+    `$ cat e2e/config.yaml.sample >> e2e/config.yaml`
+
+- Prepare config.yaml by adding kubeconfig file path, with format:
+
+    ```yaml
+    Clusters:
+        c1:
+            kubeconfigpath: /path/to/kubeconfig/c1
+        c2:
+            kubeconfigpath: /path/to/kubeconfig/c2
+        hub:
+            kubeconfigpath: /path/to/kubeconfig/hub
+    ```
+
+    If `drenv` was used to configure RDR clusters, easily add the clusters
+    kubeconfig file paths into `config.yaml` by:
+
+    `$ cat ~/.config/drenv/rdr/config.yaml >> config.yaml`
+
+***NOTE:*** To skip all tests related to certain deployers like
+**subscr, appset or disapp** at the configuration level, update config file
+with `unsupportedDeployers`, like:
+
+```yaml
+- name: cephfs
+    storageclassname: rook-cephfs
+    accessmodes: ReadWriteMany
+    unsupportedDeployers:
+        - disapp
+```
+
+## Run All E2E Tests
+
+- To run all the tests:
+
+    ```sh
+    $ cd e2e
+    $ ./run.sh
+    ```
+
+- To run tests using a custom config file:
+
+    `$ ./run.sh -config my_config.yaml`
+
+## Run Specific E2E Test Case
+
+- Refer to **list of test cases** below for a detailed breakdown of
+    available tests. Example:
+
+    `$ ./run.sh -run TestSuites/Exhaustive/subscr-deploy-rbd-busybox`
+
+    Above command runs all test steps related to subscription based RBD busybox application.
+
+- Or, run tests using RegEx:
+
+    `$ ./run.sh -run //subscr-*`
+
+    Above command runs all tests related to subscription based application
+    deployment, which includes both RBD and CephFS pvc based applications.
+
+## Run Specific E2E Test Step
+
+- Example:
+
+    `$ ./run.sh -run TestSuites/Exhaustive/appset-deploy-rbd-busybox/Deploy`
+
+    Above command only runs **Deploy** step, which just deploys busybox
+    application with RBD pvc, using ApplicationSet CR.
```

Comment:
This does not work correctly now since the channel is deleted when the deploy step ends, which is likely to break ocm and the next step. We can lave this section for later, or add a note that this is a known issue.

You can use this syntax to add notes:

    ```
    > [!IMPORTANT]
    > Description...
    ```

It becomes:

> [!IMPORTANT]
> Description...

See https://github.com/orgs/community/discussions/16925

Reply:
As discussed, removed running specific test step section for now